spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/batch/SimpleJobService.java

```diff
@@ -484,8 +456,7 @@ private List<JobExecution> getJobExecutions(String jobName, BatchStatus status,
 	}
 
 	private void checkJobExists(String jobName) throws NoSuchJobException {
-		if (getJobNames().stream().anyMatch(e -> e.contains(jobName)) ||
-				jobInstanceDao.countJobInstances(jobName) > 0) {
+		if (jobInstanceDao.countJobInstances(jobName) > 0) {
```

Comment:
I know it was like this before but I think it would be more readable if we inverted and if the count is`<= 0` throw the ex since its a void return the other branch falls off.